pkg/workceptor/command_test.go

```diff
@@ -0,0 +1,460 @@
+package workceptor_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ansible/receptor/pkg/workceptor"
+	"github.com/ansible/receptor/pkg/workceptor/mock_workceptor"
+	"github.com/golang/mock/gomock"
+)
+
+func statusExpectCalls(mockBaseWorkUnit *mock_workceptor.MockBaseWorkUnitForWorkUnit) {
+	statusLock := &sync.RWMutex{}
+	mockBaseWorkUnit.EXPECT().GetStatusLock().Return(statusLock).Times(2)
+	mockBaseWorkUnit.EXPECT().GetStatusWithoutExtraData().Return(&workceptor.StatusFileData{})
+	mockBaseWorkUnit.EXPECT().GetStatusCopy().Return(workceptor.StatusFileData{
+		ExtraData: &workceptor.CommandExtraData{},
+	})
+}
+
+func createTestSetup(t *testing.T) (workceptor.WorkUnit, *mock_workceptor.MockBaseWorkUnitForWorkUnit, *mock_workceptor.MockNetceptorForWorkceptor, *gomock.Controller, *workceptor.Workceptor) {
+	ctrl := gomock.NewController(t)
+	ctx := context.Background()
+
+	mockBaseWorkUnit := mock_workceptor.NewMockBaseWorkUnitForWorkUnit(ctrl)
+	mockNetceptor := mock_workceptor.NewMockNetceptorForWorkceptor(ctrl)
+	mockNetceptor.EXPECT().NodeID().Return("NodeID")
+
+	w, err := workceptor.New(ctx, mockNetceptor, "/tmp")
+	if err != nil {
+		t.Errorf("Error while creating Workceptor: %v", err)
+	}
+
+	cwc := &workceptor.CommandWorkerCfg{}
+	mockBaseWorkUnit.EXPECT().Init(w, "", "", workceptor.FileSystem{}, nil)
+	workUnit := cwc.NewWorker(mockBaseWorkUnit, w, "", "")
+
+	return workUnit, mockBaseWorkUnit, mockNetceptor, ctrl, w
+}
+
+func TestCommandSetFromParams(t *testing.T) {
+	wu, mockBaseWorkUnit, _, _, _ := createTestSetup(t) //nolint:dogsled
```

Comment:
interesting lint error, any idea why this has appeared? Will pinning the linter to a version help?

Reply:
Yeah the lint error was due to have 3 blank vars returning from that func. Now that I have eliminated `ctrl` from the return. I will see if that still persists

Reply:
Yeah it looks like it worked with just having two blank vars.